### ISSUE TITLE
A single model multivariate forecaster

### ISSUE THREAD

**Issue by KishManani** [JoaquinAmatRodrigo/skforecast#626]
Currently in `ForecastMultivariate` a separate model is trained for each level/variable. Would it be possible to train a single model across all variables where an indicator variable (e.g., one hot encoding, ordinal encoding, etc.) is used to tell the model which variable it is forecasting (similar to what is currently done in `ForecasterMultiSeries`)? I've not had any experience with such an approach and was wondering whether you have any thoughts on this approach.

Thanks again,
Kishan

**Comment by JavierEscobarOrtiz** [comment]
Hello Kishan,

As it is now, the ForecasterAutoregMultiVariate is only trained in one level (series 1 or series 2) and creates one model per step (direct approach). In the figure, let's say that if you specify level = 'Series 1' when creating the Forecaster, it will only use the training matrix at the top.

![forecaster_multivariate_train_matrix_diagram](https://github.com/JoaquinAmatRodrigo/skforecast/assets/95030312/dbf1231b-178f-4939-9695-2b754e492d21)

Training a single model across all variables as in `ForecasterMultiseries` is not possible because the training matrix grows horizontally (not vertically as in the MultiSeries approach) and you will have multiple columns to use as response variables.

Best,
Javi

**Comment by JoaquinAmatRodrigo** [comment]
Hi,
I think the type of forecaster @KishManani mentions can be created with two approaches:

1. Using a multi-output regressor (multi-target) from [sklearn.multioutput](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.multioutput.MultiOutputRegressor)

2. Using a regressor that natively allows multioutput (multi-target)

The second approach is the one where neural network architectures can help. In the next releases (0.12.0) we will add a new forecaster [ForecasterRNN](https://github.com/JoaquinAmatRodrigo/skforecast/blob/0.12.x/skforecast/ForecasterRnn/ForecasterRnn.py) that will allow using Keras models within the skforecast framework, including the multi-series-multistep scenario. We are currently writing the documentation, but the code is already available.

@JavierEscobarOrtiz and @fernando-carazo Let's investigate this further to see if we can extend the modeling approaches.

**Comment by lizhuoq** [comment]
@KishManani I don't think this approach is feasible because tree models rely on a recursive process of splitting points to partition the tree into subtrees. Predictions are made by summing the samples at the leaf nodes. If the numerical ranges of different time series vary greatly, this can lead to errors. The only effective approach would be to split the tree models based on series ID in the first split, which is equivalent to modeling different series IDs separately.
